GetDKAN/dkan
# EntityStorageInterface::loadRevision is deprecated

```
  1x: The Drupal\Core\Entity\EntityStorageInterface::loadRevision method is deprecated (in drupal:10.1.0 and is removed from drupal:11.0.0. Use \Drupal\Core\Entity\RevisionableStorageInterface::loadRevision instead.).
    1x in ResourcePurgerTest::testRevisionDisappearing from Drupal\Tests\datastore\Unit\Service 
```
This is used in a number of places within DKAN core.

CRs:

https://www.drupal.org/node/2927226

https://www.drupal.org/node/3294237

Since `RevisionableStorageInterface` was added in Drupal 8.5.0, we're good on BC.

### Analysis:
Since the only place this deprecation is being thrown is `Drupal\Tests\datastore\Unit\Service\ResourcePurgerTest`, we can deduce that we're good in other places within DKAN.

`ResourcePurgerTest` uses `NodeStorageInterface` to mock node storage, which seems reasonable. However, for some reason, even though `NodeStorageInterface` inherits from `RevisionableStorageInterface`, we see this deprecation message.

So since `ResourcePurgerTest` only mocks methods of `NodeStorageInterface` which are also in `RevisionableStorageInterface`, we can mock `RevisionableStorageInterface` instead. We now don't see those deprecation messages.

### Testing:

- Use Drupal 10.1.x in a local environment.
- Configure PHPUnit to use symfony/phpunit-bridge to report deprecations.
  - Add listener to phpunit.xml file.
  - Configure SYMFONY_DEPRECATIONS_HELPER to be max[self]=0
  - Docs: https://symfony.com/doc/current/components/phpunit_bridge.html#installation
- Run tests.
- Verify that no deprecation messages related to EntityStorageInterface arise.
